caddyhttp/proxy/reverseproxy.go

```diff
@@ -331,51 +340,148 @@ type connHijackerTransport struct {
 }
 
 func newConnHijackerTransport(base http.RoundTripper) *connHijackerTransport {
-	transport := &http.Transport{
-		Proxy: http.ProxyFromEnvironment,
-		Dial: (&net.Dialer{
-			Timeout:   30 * time.Second,
-			KeepAlive: 30 * time.Second,
-		}).Dial,
-		TLSHandshakeTimeout: 10 * time.Second,
+	t := &http.Transport{
 		MaxIdleConnsPerHost: -1,
 	}
-	if base != nil {
-		if baseTransport, ok := base.(*http.Transport); ok {
-			transport.Proxy = baseTransport.Proxy
-			transport.TLSClientConfig = baseTransport.TLSClientConfig
-			transport.TLSHandshakeTimeout = baseTransport.TLSHandshakeTimeout
-			transport.Dial = baseTransport.Dial
-			transport.DialTLS = baseTransport.DialTLS
-			transport.MaxIdleConnsPerHost = -1
+	if b, _ := base.(*http.Transport); b != nil {
```

Comment:
I'm not sure if I understand… Do you agree that the code is fine? (My English skills are seriously lacking lately 😩)

Basically what I said is that the first code snipped (the old code) above is the same as:
```go
if b, ok := base.(*http.Transport); b != nil && ok == true {
}
```

But since `b` being non-nil **must** imply that `ok` is true we can drop that check and get the latter snipped (the new code). You can try that in your Playground by making it return e.g. an `&sync.Mutex{}` instead.